tests/unit/libexpr/flake/url-name.cc

```diff
@@ -5,11 +5,13 @@ namespace nix {
 
 /* ----------- tests for url-name.hh --------------------------------------------------*/
 
-    TEST(getNameFromURL, getsNameFromURL) {
+    TEST(getNameFromURL, getNameFromURL) {
```

Comment:
The `getNameFromURL` function is only used in `nix profile` and its complexity seems like a potential liability for reproducibility.
- Does this change break existing profile manifests?
- Could you move it into `profile.cc`?

Reply:
Moving it there could be a logic next step of getting it out of `libutil` as I proposed, but then we will have a hard time unit testing it if we still want to?

Reply:
I would, however, not mind a libprofile with all things profile related!

Reply:
Short of having a library for each group of commands, we could compile the CLI as a static library, so that we can at least unit test anything.
While there's something to be said for exposing all logic to potential library users, I also think that decisions about public interfaces shouldn't block testing, which is more of a basic necessity.

Reply:
Maybe `getNameFromURL` should be moved, but that's orthogonal to this PR since this function already existed.